File: gangs-of-four/prototype/samples/prototype.cs
using System;
 
public class Button
{
    public string Title { get; set; }
    public string Color { get; set; }
    public double StrokeWidth { get; set; }

    public Button(string title, string color, double strokeWidth)
    {
        Title = title;
        Color = color;
        StrokeWidth = strokeWidth;
    }

    public Button CopyWith(string title = null, string color = null, double? strokeWidth = null)
    {
        return new Button(
            title ?? this.Title,
            color ?? this.Color,
            strokeWidth ?? this.StrokeWidth
        );
    }
    
    public void PrintInfo()
    {
        Console.WriteLine("Title: " + Title + ", Color: " + Color + ", Stroke Width: " + StrokeWidth);
    }
}

public class Program
{
    public static void Main()
    {
        Button originalButton = new Button("Original", "blue", 2.0);
        Button newButton = originalButton.CopyWith("New Title");

        originalButton.PrintInfo();
        newButton.PrintInfo();
    }
}

File: gangs-of-four/prototype/samples/prototype.dart
// dart prototype.dart

class Button {
  final String title;
  final String color;
  final double strokeWidth;

  Button({
    required this.title,
    required this.color,
    required this.strokeWidth,
  });

  Button copyWith({
    String? title,
    String? color,
    double? strokeWidth,
  }) {
    return Button(
      title: title ?? this.title,
      color: color ?? this.color,
      strokeWidth: strokeWidth ?? this.strokeWidth,
    );
  }
  
  void printInfo() {
    print('Title: $title, Color: $color, Stroke With: $strokeWidth');
  }
}

void main() {
  Button originalButton = Button(
    title: 'Original',
    color: '#0000FF',
    strokeWidth: 2.0,
  );
  Button newButton = originalButton.copyWith(title: 'New Title');

  originalButton.printInfo();
  newButton.printInfo();
}